lib/dry/transaction/step.rb

```diff
@@ -27,12 +29,24 @@ def initialize(step_adapter, step_name, operation_name, operation, options, call
         @call_args = call_args
       end
 
-      def with_call_args(*call_args)
-        self.class.new(step_adapter, step_name, operation_name, operation, options, call_args, &block)
+      def curry(operation: UNDEFINED, call_args: UNDEFINED)
```

Comment:
I think my TODO about renaming this to `#curry` was actually out of date. I don't think curry as a name makes sense if we're passing more than just call args. Would you mind renaming again to `#with`? That's more consistent with similar methods elsewhere in dry-rb land.